### DISCUSSION TITLE
How to setup CPU Temperature Monitoring

### DISCUSSION THREAD

**Discussion by cochraneserver** [netdata/netdata#13265] [Q&A] [answered]
Hi,

I have netdata agent setup on Proxmox, sendmail is configured and test notifications via e-mail work. I'm trying to get CPU core temperature alerts to work but looks like I need some help.

	sensors
	
		acpitz-acpi-0
		Adapter: ACPI interface
		temp1:        +27.8°C  (crit = +105.0°C)
		temp2:        +29.8°C  (crit = +105.0°C)
		
		coretemp-isa-0000
		Adapter: ISA adapter
		Package id 0:  +31.0°C  (high = +80.0°C, crit = +100.0°C)
		Core 0:        +27.0°C  (high = +80.0°C, crit = +100.0°C)
		Core 1:        +30.0°C  (high = +80.0°C, crit = +100.0°C)
		Core 2:        +28.0°C  (high = +80.0°C, crit = +100.0°C)
		Core 3:        +31.0°C  (high = +80.0°C, crit = +100.0°C)

I've set the warning alarm very low (15) with the purpose of triggering an alarm

    nano /etc/netdata/health.d/cpu_temp.conf
  
        alarm: cpu_temp
        on: sensors.coretemp-isa-0000_temperature
        os      : linux
        hosts   : *
        units   : Celsius
        every   : 5m
        warn: ${Core 0} > 15 OR ${Core 1} > 15 OR ${Core 02} > 15 OR ${Core 3} > 15
        crit: ${Core 0} > 65 OR ${Core 1} > 65 OR ${Core 02} > 65 OR ${Core 3} > 65
        info    : CPU temperature is high
        to      : sysadmin

The chart shows core temps are around 25 but no alarm is triggering. I have restarted netdata service and also ran netdatacli reload-health.

![image](https://user-images.githubusercontent.com/108437162/176559652-2a265ef1-fdff-47a5-a6d2-82b9c3475727.png)

Looking forward to any input - not very familiar with netdata unfortunately but I am about 90% done and would like to settle on it for Proxmox and then TrueNAS Core.

Thanks

**Comment by thiagoftsm** [comment, marked as answer]
Hello @cochraneserver ,

If you want create an [alert](https://learn.netdata.cloud/docs/monitor/configure-alarms) for all cores, probably it is better to use [foreach](https://learn.netdata.cloud/docs/agent/health/reference#example-5---cpu-usage) in your lookup instead to define one alert per dimension, so your alert would look like something like this:

```
alarm: cpu_temp
    on: sensors.coretemp-isa-0000_temperature
    os      : linux
    hosts   : *
   lookup: average -5m foreach Core*
    units   : Celsius
    every   : 5m
    warn:  $this > 15
    crit:  $this > 65
    info    : CPU temperature is high
    to      : sysadmin
```

Any doubt, please, let us know! :)

Best regards!

**Comment by cochraneserver** [comment]
This worked perfecty, thank you so much.
